src/common/meta/src/key/topic_region.rs

```diff
@@ -0,0 +1,228 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::{self, Display};
+
+use serde::{Deserialize, Serialize};
+use snafu::OptionExt;
+use store_api::storage::RegionId;
+
+use crate::error::{Error, InvalidMetadataSnafu, Result};
+use crate::key::{MetadataKey, TOPIC_REGION_PATTERN, TOPIC_REGION_PREFIX};
+use crate::kv_backend::KvBackendRef;
+use crate::rpc::store::{BatchPutRequest, PutRequest, RangeRequest};
+use crate::rpc::KeyValue;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct TopicRegionKey<'a> {
+    pub region_id: RegionId,
+    pub topic: &'a str,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct TopicRegionValue;
+
+impl<'a> TopicRegionKey<'a> {
+    pub fn new(region_id: RegionId, topic: &'a str) -> Self {
+        Self { region_id, topic }
+    }
+
+    pub fn range_start_key() -> String {
+        TOPIC_REGION_PREFIX.to_string()
+    }
+
+    pub fn range_topic_key(topic: &str) -> String {
+        format!("{}/{}", TOPIC_REGION_PREFIX, topic)
+    }
+}
+
+impl<'a> MetadataKey<'a, TopicRegionKey<'a>> for TopicRegionKey<'a> {
+    fn to_bytes(&self) -> Vec<u8> {
+        self.to_string().into_bytes()
+    }
+
+    fn from_bytes(bytes: &'a [u8]) -> Result<TopicRegionKey<'a>> {
+        let key = std::str::from_utf8(bytes).map_err(|e| {
+            InvalidMetadataSnafu {
+                err_msg: format!(
+                    "TopicRegionKey '{}' is not a valid UTF8 string: {e}",
+                    String::from_utf8_lossy(bytes)
+                ),
+            }
+            .build()
+        })?;
+        TopicRegionKey::try_from(key)
+    }
+}
+
+impl Display for TopicRegionKey<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "{}/{}/{}",
+            TOPIC_REGION_PREFIX,
+            self.topic,
+            self.region_id.as_u64()
```

Comment:
```suggestion
            "{}/{}",
            Self::range_topic_key(self.topic),
            self.region_id.as_u64()
```

If we need to change the `range topic key` in the future, it's best to ensure that it only needs to be modified in one place.